### PR TITLE
Closes #183 | Implement `wongnai_reviews` dataloader

### LINKED ISSUE #183: Create dataset loader for Wongnai Reviews
Dataloader name: `wongnai_reviews/wongnai_reviews.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?wongnai_reviews

|     Dataset| wongnai_reviews |
|-------------|---|
| Description | Wongnai features over 200,000 restaurants, beauty salons, and spas across Thailand on its platform, with detailed information about each merchant and user reviews. Its over two million registered users can search for what’s top rated in Bangkok, follow their friends, upload photos, and do quick write-ups about the places they visit.  Each write-up(review) also comes with a rating score ranging from 1-5 stars. The task here is to create a rating prediction model using only textual information. |
| Subsets | - |
| Languages | tha |
| Tasks | Text Classification |
| License | GNU Lesser General Public License v3.0 (lgpl-3.0) |
| Homepage | https://github.com/wongnai/wongnai-corpus |
| HF URL | https://huggingface.co/datasets/wongnai_reviews |
| Paper URL | https://github.com/wongnai/wongnai-corpus |

### PR DESCRIPTION
**Pull request by chenxwh**
Closes #183.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_text_multi",
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py:57]: following the task type, the schema (and therefore the config name) should be `seacrowd_text`. This also applies here and throughout the code.

File: seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_text_multi",
+            subset_id=_DATASETNAME,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py:68]:
```suggestion
        star_rating = ["1", "2", "3", "4", "5"]
        if self.config.schema == "source":
```

File: seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            features = datasets.Features(
+                {
+                    "review_body": datasets.Value("string"),
+                    "star_rating": datasets.ClassLabel(names=["1", "2", "3", "4", "5"]),
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py:72]:
```suggestion
                    "star_rating": datasets.ClassLabel(names=star_rating),
```

File: seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                }
+            )
+
+        elif self.config.schema == "seacrowd_text_multi":
+            features = schemas.text_multi_features(["1", "2", "3", "4", "5"])
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py:77]:
```suggestion
            features = schemas.text_features(label_names=star_rating)
```

File: seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+_LANGUAGES = ["tha"]
+
+_LICENSE = Licenses.LGPL_3_0.value
+
+_LOCAL = False
+
+_URLS = {_DATASETNAME: "https://archive.org/download/wongnai_reviews/wongnai_reviews_withtest.zip"}
+
+_SUPPORTED_TASKS = [Tasks.ASPECT_BASED_SENTIMENT_ANALYSIS]
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py:35]:
```suggestion
_SUPPORTED_TASKS = [Tasks.TEXT_CLASSIFICATION]
```

File: seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+            with open(filepath, encoding="utf-8") as f:
+                spamreader = csv.reader(f, delimiter=";", quotechar='"')
+                for i, row in enumerate(spamreader):
+                    yield i, {"review_body": row[0], "star_rating": row[1]}
+
+        elif self.config.schema == "seacrowd_text_multi":
+            with open(filepath, encoding="utf-8") as f:
+                spamreader = csv.reader(f, delimiter=";", quotechar='"')
+                for i, row in enumerate(spamreader):
+                    yield i, {"id": str(i), "text": row[0], "labels": [row[1]]}
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py:114]:
```suggestion
                    yield i, {"id": str(i), "text": row[0], "label": row[1]}
```

File: seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "review_body": datasets.Value("string"),
+                    "star_rating": datasets.ClassLabel(names=["1", "2", "3", "4", "5"]),
+                }
+            )
+
+        elif self.config.schema == "seacrowd_text_multi":
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py:76]:
```suggestion
        elif self.config.schema == "seacrowd_text":
```

File: seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py (hunk 1 above)
Comment (jen-santoso) [inline seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py:57]:
```suggestion
            name=f"{_DATASETNAME}_seacrowd_text",
```

File: seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_text_multi",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_text_multi",
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py:60]:
```suggestion
            schema="seacrowd_text",
```

File: seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        if self.config.schema == "source":
+            with open(filepath, encoding="utf-8") as f:
+                spamreader = csv.reader(f, delimiter=";", quotechar='"')
+                for i, row in enumerate(spamreader):
+                    yield i, {"review_body": row[0], "star_rating": row[1]}
+
+        elif self.config.schema == "seacrowd_text_multi":
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py:110]:
```suggestion
        elif self.config.schema == "seacrowd_text":
```

**Review by jen-santoso — CHANGES_REQUESTED** [review]
Hi @chenxwh, thanks for implementing the dataloader. 

The Task is listed as `Text classification` in the issue card.
Is there a case where multiple classes exist? 
If no, please change the `_SUPPORTED_TASKS` to `Tasks.TEXT_CLASSIFICATION`, along with the other suggestions here. 
Otherwise, please consider only the suggestions in line 68, 72, and 76.

**Comment by chenxwh** [comment]
Thanks for the suggestions! Made suitable changes for the new task class.

**Review by jen-santoso — APPROVED** [review]

**Comment by holylovenia** [comment]
Replacing @danjohnvelasco with @MJonibek.

**Review by MJonibek — APPROVED** [review]
@chenxwh thank you for working on this dataloader. Checked, LGTM.

**Pull request merged**